File: src/ch2d/bindings/texture.cpp
// ch2d
#include <ch2d/System.hpp>

namespace ch2d
{
    unsigned int System::texture_create(void)
    {
        unsigned int id = mTextureHandler.create();

        return id;
    }

    bool System::texture_loadFromFile(unsigned int id, std::string filename)
    {
        auto texture = mTextureHandler.get(id);

        if(nullptr == texture)
        {
            return false;
        }

        if(!texture->loadFromFile(filename))
        {
            return false;
        }

        return true;
    }

    bool System::texture_setRepeated(unsigned int id, bool repeated)
    {
        auto texture = mTextureHandler.get(id);

        if(nullptr == texture)
        {
            return false;
        }

        texture->setRepeated(repeated);

        return true;
    }

    bool System::texture_setSmooth(unsigned int id, bool smooth)
    {
        auto texture = mTextureHandler.get(id);

        if(nullptr == texture)
        {
            return false;
        }

        texture->setSmooth(smooth);

        return true;
    }

    bool System::texture_isRepeated(unsigned int id)
    {
        auto texture = mTextureHandler.get(id);

        if(nullptr == texture)
        {
            return false;
        }

        return texture->isRepeated();
    }

    bool System::texture_isSmooth(unsigned int id)
    {
        auto texture = mTextureHandler.get(id);

        if(nullptr == texture)
        {
            return false;
        }

        return texture->isSmooth();
    }

    bool System::texture_remove(unsigned int id)
    {
        return mTextureHandler.remove(id);
    }
}

File: lua/ch2d/lib/texture.lua
local Texture = {}
Texture.__index = Texture

function Texture.create()
    local texture = {}
    setmetatable(texture, Texture)

    texture.id = ch2d.texture.create()

    return texture
end

function Texture:remove()
    return ch2d.texture.remove(self.id)
end

function Texture:loadFromFile(filename)
    return ch2d.texture.loadFromFile(self.id, filename)
end

function Texture:setRepeated(repeated)
    return ch2d.texture.setRepeatead(self.id, repeated)
end

function Texture:setSmooth(smooth)
    return ch2d.texture.setSmooth(self.id, smooth)
end

function Texture:isRepeated()
    return ch2d.texture.isRepeated(self.id)
end

function Texture:isSmooth()
    return ch2d.texture.isSmooth(self.id)
end

return Texture
